File: JavaScript/2017-Grid Game.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var gridGame = function(grid) {
    let res = Number.POSITIVE_INFINITY;
    let top = grid[0].reduce((acc, x)=>acc+x), bottom = 0;
    
    for (let i=0; i<grid[0].length; i++){
        top -= grid[0][i];
        res = Math.min(res, Math.max(top, bottom));
        bottom += grid[1][i];
    }
    
    return res;
};

File: C++/2017-Grid Game.cpp
using LL = long long;
class Solution {
public:
    long long gridGame(vector<vector<int>>& grid) {
        LL firstRow = accumulate(begin(grid[0]), end(grid[0]), 0LL);
        LL res = LLONG_MAX/2, secondRow = 0;

        for (int i=0; i<grid[0].size(); i++) {
            firstRow -= grid[0][i];
            res = min(res, max(firstRow, secondRow));
            secondRow += grid[1][i];
        }

        return res;
    }
};